packages/web3-provider/src/subproviders/EthGasStationProvider.js

```diff
@@ -5,6 +5,24 @@ const CACHE_TIME = 15000
 const GAS_STATION_URL = 'https://ethgasstation.info/json/ethgasAPI.json'
 const GAS_PRICE_KEY = process.env.GAS_PRICE_KEY || 'average' // 'safeLow'
 
+/**
+ * Wraps a fetch with a timeout
+ *
+ * @param url {string} URL to fetch
+ * @param ms {number} timeout in milliseconds
+ * @returns {object} fetch response object
+ */
+function timeoutFetch(url, ms = 10000) {
+  return Promise.race([
+    fetch(url).then(res => {
```

Comment:
I think you can just say `fetch(url)` here and skip out the then

Reply:
The relationship of `Promise` and `await` still trips me up.  Would `await` resolve nested promises?  Or would `Promise.race()`?

Reply:
I think you can either `await Promise.race(promise1, promise2)` or `Promise.race(promise1, promise2).then(result => doSomething())`

Reply:
I did a quick test to be sure and it looks like await resolves nested Promises.  Stripped off that `then()`.